### ISSUE TITLE
R package for limma

### ISSUE THREAD

**Issue by peterjc** [galaxyproject/tools-iuc#122]
I noticed that I didn't finish nicely packaging this tool of mine (despite using it in published workflows):

* https://toolshed.g2.bx.psu.edu/view/peterjc/venn_list/
* https://testtoolshed.g2.bx.psu.edu/view/peterjc/venn_list/
* https://github.com/peterjc/pico_galaxy/tree/master/tools/venn_list

It needs the R package limma for plotting (and currently uses rpy too, which is another issue), which is not yet in the ToolShed as an IUC package. However, I spotted this in the Test Tool Shed (only):

* https://testtoolshed.g2.bx.psu.edu/view/tmcgowan/limma

Can we (the IUC) have an official limma package?

**Comment by erasche** [comment]
@peterjc sorry for delay, sure. I'll add it today.

**Comment by erasche** [comment]
You need a specific version? If not I'm taking 3.9.9 from http://bioarchive.s3-website-us-east-1.amazonaws.com/limma/

**Comment by erasche** [comment]
@peterjc specific version of R you want the package tied to? (Ugh, I hate that we can't say package_r>2.15)

**Comment by bgruening** [comment]
@peterjc can you use rpy2 instead? I know I have some working packages for rpy2 in the TS, used by https://github.com/bgruening/galaxytools/tree/master/tools/rpy_statistics_collection

**Comment by peterjc** [comment]
I think limma is pretty stable - at least for the Venn diagrams I am not aware of any version specific changes. So anything really.

Regarding rpy/rpy2, yes, I'm aware ``venn_list`` would need updating at some point - its just a case finding time to do it.

**Comment by erasche** [comment]
@peterjc pushed some WIP, waiting on bioarchive/aRchive_source_code#9

**Comment by peterjc** [comment]
Thanks :)

**Comment by erasche** [comment]
@peterjc okay, aRchive is ready for use now. All you need to do is in your tool_dependencies.xml file:

```
            <actions>
                <action type="setup_r_environment">
                    <repository name="package_r_3_1_2" owner="iuc">
                        <package name="R" version="3.1.2" />
                    </repository>
                    <package>http://bioarchive.s3.amazonaws.com/limma_3.25.3.tar.gz</package>
                </action>
            </actions>
```

And just use packages from http://bioarchive.s3.amazonaws.com/ in the future. You'll need to do manual dependency resolution for the moment, I believe @nitesh1989 is working on a tool to figure out the whole proper R dependency chain.

**Comment by bgruening** [comment]
@nitesh1989 you might want to look at: https://github.com/bgruening/notebooks/tree/master/R

**Comment by nitesh1989** [comment]
Cheers @bgruening @erasche

**Comment by bgruening** [comment]
@nitesh1989 seriously, with this script it should be possible to generate a complete `tool_dependency.xml` file. This would be a really nice feature to of aRchive! Let me know if you need help with this.

**Comment by erasche** [comment]
@bgruening we should finish up either a webservice/docker commands/etc in order to provide planemo with a "dependencies_init" command of some sort.

**Comment by erasche** [comment]
#147 ready.

**Comment by yhoogstrate** [comment]
line 4:   <repository name="package_r_3_1_2" owner="devteam" ...
line 11:   <repository name="package_r_3_1_2" owner="iuc">

I think the owner at line 4 should by iuc as well?

When I tried installing edgeR via this limma installation on a local Galaxy installation, some functions seem to be missing:

camera
contrastAsCoef
goana
subsetListOfArrays
tricubeMovingAverage

**Comment by erasche** [comment]
Confirmed as an issue. CC @nitesh1989 

```
** inst
** preparing package for lazy loading
Error : objects ‘camera’, ‘contrastAsCoef’, ‘goana’, ‘subsetListOfArrays’, ‘tricubeMovingAverage’ are not exported by 'namespace:limma'
ERROR: lazy loading failed for package ‘edgeR’
* removing ‘/home/users/cpt/cpt/esr/R/x86_64-pc-linux-gnu-library/3.0/edgeR’
Warning message:
In install.packages("edgeR_3.9.9.tar.gz") :
  installation of package ‘edgeR_3.9.9.tar.gz’ had non-zero exit status
> 
```

**Comment by erasche** [comment]
```
esr@cpt:~/Projects/galaxy/tools-iuc⟫ tar xfz limma_3.9.9.tar.gz
esr@cpt:~/Projects/galaxy/tools-iuc⟫ grep camera -R limma/
```

edgeR obviously expects some functions/objects from limma but they just aren't there!

**Comment by erasche** [comment]
@yhoogstrate okay, this was an out-of-date limma issue. Upgrading to limma_3.25.3.tar.gz fixes this.

**Comment by nitesh1989** [comment]
@erasche Is this solved? And is edgeR_3.9.9.tar.gz coming from aRchive?

**Comment by erasche** [comment]
@nitesh1989 yep, everything is good!

fre. 22. mai 2015 kl. 12.50 skrev Nitesh Turaga <notifications@github.com>:

**Comment by nitesh1989** [comment]
Very cool. Happy its of use!
